### PR TITLE
CATROID-272 refactor PhysicsBricksCloneTest

### PR DESCRIPTION
**Pull request by gPathpp**

### CODE REVIEW

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	public void testBrickFieldValidValue() throws InterpretationException {
+		assertSame(BRICK_FORMULA_VALUE, brickFormula.interpretInteger(sprite));
+		assertSame(BRICK_FORMULA_VALUE, cloneBrickFormula.interpretInteger(sprite));
+	}
+
+	@Test
+	public void testBrickFieldEquals() throws InterpretationException {
+		assertSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));
+	}
+}
```
Comment (DinosaurierRex) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java]: This test does kinda the same thing like the test above, but if there are invalid values in the bricks, this test may not fail, if the invalid value was correctly cloned

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	public void testChangeBrickField() throws InterpretationException {
+		cloneBrickFormula.setRoot(new FormulaElement(FormulaElement.ElementType.NUMBER, CLONE_BRICK_FORMULA_VALUE, null));
+		assertNotSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));
+	}
+
+	@Test
+	public void testBrickFieldValidValue() throws InterpretationException {
+		assertSame(BRICK_FORMULA_VALUE, brickFormula.interpretInteger(sprite));
+		assertSame(BRICK_FORMULA_VALUE, cloneBrickFormula.interpretInteger(sprite));
+	}
```
Comment (DinosaurierRex) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java]: if you delete second line here, you can add the assert from testBrickFieldEquals to do the same thing but in one
`assertSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));`
So you are interpreting the correct value first, and then check the cloned one

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/CloneSetPhysicsObjectTypeBrickTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	@Test
+	public void testSameInstance() {
+		assertThat(clonedBrick, is(instanceOf(brick.getClass())));
+	}
+
+	@Test
+	public void testOriginalAndCloneNotSame() {
+		assertNotSame(clonedBrick, brick);
+	}
+}
```
Comment (DinosaurierRex) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/CloneSetPhysicsObjectTypeBrickTest.java]: Like that one!

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/CloneWhenBounceOffBrick.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	public void testCloneWhenBounceOffBrick() throws Exception {
+		Brick brick = new WhenBounceOffBrick(new WhenBounceOffScript(null));
+		Brick clonedBrick = brick.clone();
+
+		WhenBounceOffScript brickReceiverScript = (WhenBounceOffScript) brick.getScript();
+		WhenBounceOffScript clonedBrickReceiverScript = (WhenBounceOffScript) clonedBrick.getScript();
+
+		assertNotSame(brickReceiverScript, clonedBrickReceiverScript);
+	}
+}
```
Comment (DinosaurierRex) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/CloneWhenBounceOffBrick.java]: This test can be deleted see [CloneScriptBrickTest](https://github.com/Catrobat/Catroid/blob/develop/catroid/src/test/java/org/catrobat/catroid/test/content/bricks/CloneScriptBrickTest.java)

**Comment by wslany** [comment]
Thanks!

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	public void setUp() throws CloneNotSupportedException {
+		FormulaBrick cloneBrick = (FormulaBrick) brick.clone();
+		brickFormula = brick.getFormulaWithBrickField(brickField);
+		cloneBrickFormula = cloneBrick.getFormulaWithBrickField(brickField);
+	}
+
+	@Test
+	public void testChangeBrickField() throws InterpretationException {
+		cloneBrickFormula.setRoot(new FormulaElement(FormulaElement.ElementType.NUMBER, CLONE_BRICK_FORMULA_VALUE, null));
+		assertNotSame(brickFormula.interpretInteger(sprite), cloneBrickFormula.interpretInteger(sprite));
```
Comment (84n4n4) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java]: assertNotSame != !assertEquals (best as assertThat(bla, is(not(blu)))

File: catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	@Test
+	public void testBrickFieldValidValue() throws InterpretationException {
+		assertSame(BRICK_FORMULA_VALUE, brickFormula.interpretInteger(sprite));
```
Comment (84n4n4) [inline catroid/src/test/java/org/catrobat/catroid/test/physics/clone/ClonePhysicsBricksWithFormulaTest.java]: assertEquals

**Pull request merged**